app/assets/javascripts/discourse/app/widgets/widget.js

```diff
@@ -323,6 +323,13 @@ export default class Widget {
     });
   }
 
+  callWidgetFunction(name, param) {
+    const widget = this._findAncestorWithProperty(name);
+    if (widget) {
+      return widget[name].call(widget, param);
+    }
+  }
+
   sendWidgetAction(name, param) {
     return this.rerenderResult(() => {
       const widget = this._findAncestorWithProperty(name);
```

Comment:
Shouldn't this method use the new `callWidgetFunction`?

Reply:
Yes, maybe... In one hand you DRY on the other hand you avoid some indirection of if call had found a widget then return else... Not sure it's very worth it.

Reply:
I made the change 👨‍🎨